File: src/main/java/net/teamhollow/direbats/world/DirebatsGameRules.java
package net.teamhollow.direbats.world;

import net.fabricmc.fabric.api.gamerule.v1.CustomGameRuleCategory;
import net.fabricmc.fabric.api.gamerule.v1.GameRuleFactory;
import net.fabricmc.fabric.api.gamerule.v1.GameRuleRegistry;
import net.minecraft.text.TranslatableText;
import net.minecraft.util.Formatting;
import net.minecraft.util.Identifier;
import net.teamhollow.direbats.Direbats;

import static net.minecraft.world.GameRules.*;

public class DirebatsGameRules {
    public static final CustomGameRuleCategory CATEGORY = new CustomGameRuleCategory(
        new Identifier(Direbats.MOD_ID, "category"),
        new TranslatableText("gamerule.%s.category".formatted(Direbats.MOD_ID))
            .formatted(Formatting.YELLOW, Formatting.BOLD)
    );

    public static final Key<BooleanRule> DIREBAT_ITEM_PICKUP = register("direbatItemPickup", true);

    private static <T extends Rule<T>> Key<T> register(String id, Type<T> type) {
        return GameRuleRegistry.register("%s:%s".formatted(Direbats.MOD_ID, id), DirebatsGameRules.CATEGORY, type);
    }

    private static Key<BooleanRule> register(String id, boolean defaultValue) {
        return register(id, GameRuleFactory.createBooleanRule(defaultValue));
    }
}

File: src/main/kotlin/dev/andante/direbats/world/DirebatsGameRules.kt
package dev.andante.direbats.world

import dev.andante.direbats.Direbats
import net.fabricmc.fabric.api.gamerule.v1.GameRuleFactory
import net.fabricmc.fabric.api.gamerule.v1.GameRuleRegistry
import net.minecraft.world.GameRules
import net.minecraft.world.GameRules.BooleanRule
import net.minecraft.world.GameRules.Category

/**
 * Represents Direbats game rules.
 */
object DirebatsGameRules {
    /**
     * Whether or not Direbats can locate and loot item entities.
     */
    val DO_DIREBAT_ITEM_PICKUP: GameRules.Key<BooleanRule> =
        GameRuleRegistry.register("${Direbats.MOD_ID}:doDirebatItemPickup",
            Category.MOBS, GameRuleFactory.createBooleanRule(true)
        )
}
